app/models/concerns/order_summary.rb

```diff
@@ -0,0 +1,25 @@
+module OrderSummary
+  extend ActiveSupport::Concern
+
+  def subtotal
+    order_items.sum(&:total_price)
+  end
+
+  def discount
+    coupon.try(:discount) || 0.00
+  end
+
+  def delivery_price
+    delivery.try(:price) || 0.00
+  end
+
+  def order_total
+    subtotal - discount + delivery_price
+  end
+
+  def items_count
+    count = order_items.collect(&:quantity).compact.sum
+    return false if count.zero?
```

Comment:
I think it is not readable when method `items_count` return false. According to name this method must return integer.